src/libsystemd-network/ndisc-protocol.c

```diff
@@ -0,0 +1,30 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "ndisc-protocol.h"
+
+int pref64_plc_to_prefix_length(uint8_t plc, uint8_t *ret) {
+        switch (plc) {
+        case PREF64_PLC_32:
```

Comment:
Maybe, better to introduce enum and a table:
```c
typedef enum PrefixLengthCode {
        PREFIX_LENGTH_CODE_96 = 0,
        ...
        _PREFIX_LENGTH_CODE_MAX,
        _PREFIX_LENGTH_CODE_INVALID = -EINVAL,
} PrefixLengthCode;

static const uint8_t prefix_length_code_to_prefix_length[_PREFIX_LENGTH_CODE_MAX] = {
        [PREFIX_LENGTH_CODE_96] = 96,
        ...
};

int pref64_plc_to_prefix_length(uint16_t plc, uint8_t *ret) {
        assert(ret);

        plc &= PREF64_PLC_MASK;
        if (plc >= _PREFIX_LENGTH_CODE_MAX)
                return -EINVAL;

        *ret = prefix_length_code_to_prefix_length[plc];
        return 0;
}
```
At least, please add an assertion for `ret`.